src/plugins/api_tests/src/core/util/MsaUtilsUnitTests.cpp

```diff
@@ -97,4 +101,23 @@ IMPLEMENT_TEST(MsaUtilsUnitTests, all_names_with_spaces) {
     CHECK_NO_ERROR(os);
 }
 
+IMPLEMENT_TEST(MsaUtilsUnitTests, get_len_with_big_args) {
+    U2OpStatusImpl os;
+
+    DynTable dynTable(4733, 4733, false);
+    int newLen = dynTable.getLastLen();
+    CHECK_NO_ERROR(os);
+    CHECK_EQUAL(4733, newLen, "getLastLen() return value");
+
+    DynTable dynTable2(3733, 4733, false);
+    newLen = dynTable2.getLastLen();
+    CHECK_NO_ERROR(os);
+    CHECK_EQUAL(3733, newLen, "getLastLen() return value");
+
+    DynTable dynTable3(44733, 44733, false);
```

Comment:
How this check is different from the first one?

Can you please add more tests for a small table with custom paths inside of the table. 
All 3 tests you added check only degenerated case of empty table, right?

Reply:
The test checks the long loops, which were failed on the Mac
All other real search are in xml tests.
There are more then nine thousands calls of the method
Do you really need to repeat the same calls (maybe with different args) ?

Reply:
Right, it is enough a single test to check the long loop problem on Mac OS and you already have a regression test for that. You can add 1 more XML - this is OK too.

But you have rewritten the whole method: the method depends on the internal matrix content: it computes the result length using content of the matrix.

The tests you added do not deal with it: you always fill the matrix with 0 values. Could you please add tests with some meaningful and non-degenerated content?